File: frontend/src/types/content.ts
export type Platform = "facebook" | "instagram" | "linkedin" | "all";

export type ContentType = "post" | "reel" | "carousel" | "story";

export type ContentGenerationRequest = {
  businessName: string;
  targetAudience: string;
  topic: string;
  platform: Platform;
  contentType: ContentType;
};

export type GeneratedContent = {
  post: string;
  reelScript?: string;
  caption: string;
  hashtags: string[];
  callToAction: string;
};

export type ContentGenerator = {
  generate: (request: ContentGenerationRequest) => GeneratedContent;
};

File: backend/src/invictus_os/schemas/content.py
from typing import Literal

from pydantic import BaseModel, Field

Platform = Literal["facebook", "instagram", "linkedin", "all"]
ContentType = Literal["post", "reel", "carousel", "story"]


class ContentGenerationRequest(BaseModel):
    business_name: str = Field(min_length=1, max_length=120)
    target_audience: str = Field(min_length=1, max_length=240)
    topic: str = Field(min_length=1, max_length=500)
    platform: Platform
    content_type: ContentType


class GeneratedContentResponse(BaseModel):
    post: str = Field(min_length=1)
    reel_script: str | None = None
    caption: str = Field(min_length=1)
    hashtags: list[str] = Field(min_length=1, max_length=12)
    call_to_action: str = Field(min_length=1)
